src/hooks/useOktaUserLookup.tsx

```diff
@@ -0,0 +1,75 @@
+import { useMemo, useState } from 'react';
+import { useQuery } from '@apollo/client';
+
+import SearchOktaUsers from 'queries/SearchOktaUsers';
+import { SearchOktaUsers as SearchOktaUsersType } from 'queries/types/SearchOktaUsers';
+
+/** Cedar contact properties */
+export type OktaUserType = {
+  username: string;
+  displayName: string;
+};
+
+type OktaHookProps = {
+  contacts: OktaUserType[];
+  queryOktaUsers: (searchTerm: string) => void;
+  loading: boolean;
+};
+
+/**
+ * Custom hook for retrieving contacts from Cedar by common name
+ * */
+function useOktaUserLookup(
+  query?: string | null,
+  userSelected?: boolean
+): OktaHookProps {
+  const [searchTerm, setSearchTerm] = useState<string | null | undefined>(
+    query
+  );
+
+  const { data, previousData, loading } = useQuery<SearchOktaUsersType>(
+    SearchOktaUsers,
+    {
+      variables: { searchTerm },
+      skip: !searchTerm || searchTerm.length < 3 || userSelected
+    }
+  );
+
+  /**
+   * Update search term if query is more than 2 characters long
+   */
+  const updateQuery = (nameQuery: string) => {
+    if (nameQuery.length > 1) setSearchTerm(nameQuery);
```

Comment:
NIT: It's possible that a search term could be the same causing an unnecessary re-render. For instance, if you copy-pasted the same search term I believe it would run the query. This is very unlikely and should have a low incidence rate, but if you wanted to guard against it you could do this.

```suggestion
    if (nameQuery.length > 1 && nameQuery !== searchTerm) setSearchTerm(nameQuery);
```

Reply:
I am not worried about guarding against this case for now, given the chance that this happens (and that it doesn't really do anything but make one more request), but that's an interesting catch! I think I'll resolve this one for now if that's alright!